src/libs/hovercard/hovercard.js

```diff
@@ -0,0 +1,184 @@
+
+	//show placeholder
+	const placeholder = document.createElement("img");
+	placeholder.src = runtime.getURL("icons/fennecspin.gif");
+
+	const tempHeader = createElementWithClass("div", "account__header");
+
+	tempHeader.appendChild(placeholder);
+	hovercard.appendChild(tempHeader);
+
+	//set position from mouse cursor
+	hovercard.style.left = (mousePos.x + 10).toString() + "px";
+	hovercard.style.top = (mousePos.y + 10).toString() + "px";
+
+	hovercard.classList.add("protoots-hovercard");
+	hovercard.id = "protoots-hovercard";
+
+	layer.appendChild(hovercard);
+
+	//get account
+	const { account, relationship } = await fetchProfile(statusID, accountName);
+	if (!account) return;
+
+	const settings = hoverCardSettings();
+
+	//generate the profile and add it to the card
+	const [profileElement, bio] = generateProfile(account, relationship, settings);
+	hovercard.appendChild(profileElement);
+
+	//delete placeholder
+	hovercard.removeChild(tempHeader);
+
+	//get relationship from API again and redo those elements
+	//this is because a cached relationship may easily be out of date
+	//i.e. the users sees someone new, clicks follow, which then isn't reflected in the card
+	const updatedRelationship = await fetchRelationship(account.id);
+	const [updatedProfileElement, updatedbio] = generateProfile(
+		account,
+		updatedRelationship,
+		settings,
+	);
+	hovercard.replaceChild(updatedProfileElement, profileElement);
+
+	//If bio would be scrollable, but that settings is disabled, add the button
+	if (updatedbio.scrollHeight > updatedbio.clientHeight && !settings.scrollableBio) {
+		// console.log("bio is scrollable");
+		// bio.style.overflow = "hidden";
+		updatedbio.classList.add("showmore");
+		addShowMoreButton(updatedbio, hovercard);
+	}
+
+	moveOnScreen(hovercard);
+
+	hovercard.addEventListener("mouseenter", () => (hovering = el));
+	hovercard.addEventListener("mouseleave", () => {
+		hovering = false;
+		hovercardExists = false;
+		setTimeout(() => {
+			if (hovering) return;
+			removeHoverCard(hovercard);
+		}, listenerTimeout);
+	});
+	addEventListener(
+		"scroll",
+		() => {
+			removeHoverCard(hovercard);
+		},
+		{ once: true },
+	);
+}
+
+export function moveOnScreen(hovercard) {
+	const boundingrect = hovercard.getBoundingClientRect();
+
+	const borderDistance = {
+		x: boundingrect.right - window.innerWidth,
+		y: boundingrect.bottom - window.innerHeight,
+	};
+
+	if (borderDistance.y > 0) {
+		hovercard.style.top =
+			(Number(hovercard.style.top.replace("px", "")) - borderDistance.y).toString() + "px";
+	}
+	if (borderDistance.x > 0) {
+		hovercard.style.left =
+			(Number(hovercard.style.left.replace("px", "")) - borderDistance.y).toString() + "px";
+	}
+}
+
+/**
+ *
+ * @param {HTMLElement} card Element which to remove after timeout
+ */
+function removeHoverCard(card) {
+	if (layer.getAttribute("noremove")) return; //debug settings
+	card.classList.add("hidden"); //TODO: interrupt removal by hovering again
```

Comment:
this todo statement makes even less sense after reading the setTimeout-callback's inside the mouseleave